File: CBORTest/CBORExceptionTest.cs
using System;
using NUnit.Framework;
using PeterO.Cbor;

namespace Test {
  [TestFixture]
  public class CBORExceptionTest {
    [Test]
    public void TestConstructor() {
      try {
        throw new CBORException("Test exception");
      } catch (CBORException) {
        // NOTE: Intentionally empty
      } catch (Exception ex) {
        Assert.Fail(ex.ToString());
        throw new InvalidOperationException(String.Empty, ex);
      }
    }
  }
}

File: src/test/java/com/upokecenter/test/CBORExceptionTest.java
package com.upokecenter.test;

import org.junit.Assert;
import org.junit.Test;
import com.upokecenter.cbor.*;

  public class CBORExceptionTest {
    @Test(expected = CBORException.class)
    public void TestConstructor() {
      throw new CBORException("Test exception");
    }
  }
